src/websocket/stream.py

```diff
@@ -1,113 +1,124 @@
-import asyncio
 import json
 import time
+from typing import Dict, List
 import uuid
 import traceback
+import asyncio
 
 from fastapi import WebSocket, WebSocketDisconnect
+from fastapi.websockets import WebSocketState
+from faster_whisper.transcribe import Word
 from pydub import AudioSegment
 
 from src.helper import logger
 from src.helper.data_handler import DataHandler
 from src.helper.local_agreement import LocalAgreement
-from src.helper.segment_info_parser import parse_segments_and_info_to_dict
 from src.websocket.stream_transcriber import Transcriber
 
 # To Calculate the seconds of audio in a chunk of 16000 Hz, 2 bytes per sample and 1 channel (as typically used in Whisper):
 # 16000 Hz * 2 bytes * 1 channel = 32000 bytes per second
 BYTES_PER_SECOND = 32000
 
-# This is the number of seconds we wait before printing a final transcription
-FINAL_TRANSCRIPTION_TIMEOUT = 6
+# This is the number of words we wait before printing a final transcription
+FINAL_TRANSCRIPTION_THRESHOLD = 6
+
+# Max size of window defined in bytes
+MAX_WINDOW_SIZE_BYTES = BYTES_PER_SECOND * 15
 
-# This is the number of seconds we wait before printing a partial transcription
-PARTIAL_TRANSCRIPTION_TIMEOUT = 1
+# Bytes after which a retranscription of the window is triggered
+PARTIAL_TRANSCRIPTION_BYTE_THRESHOLD = BYTES_PER_SECOND * 1
 
+# If no final has been published for this long just publish all as final
+# This is mostly for cases where no audio data is sent
+FINAL_PUBLISH_SECOND_THRESHOLD_FACTOR = 5
 
 class Stream:
     def __init__(self, transcriber: Transcriber, id: int):
-        self.logger = logger.get_logger_with_id(__name__, str(id))
+        self.logger = logger.get_logger_with_id(__name__, f"{id}")
         self.transcriber = transcriber
         self.id = id
         self.close_stream = False
 
-        # Cache for the last few chunks of audio
-        self.recently_added_chunk_cache = b""
-
-        # Cache for the chunks since the last final
-        self.chunk_cache = b""
-        # Last final bytes and text to check for duplicates and create prompts
-        self.last_final_result_object = {"text": "", "result": []}
-        self.last_final_bytes = b""
+        self.sliding_window = b""
+        self.total_bytes = b""
```

Comment:
They can grow indefinetely no?
For us maybe out of scope but noting somewhere that there maybe has to be a cleanup functionality if I am not mistaken would be good.